lib/manager.rb

```diff
@@ -0,0 +1,114 @@
+require "pry"
+require_relative "room"
+
+module Hotel
+  class Manager
+    attr_accessor :rooms_reservations_hash
+
+    def initialize
+      @rooms_reservations_hash = Room.make_rooms
+    end
+
+    # Manager.rooms_res_hash builds this structure:
+    # {
+    #     1:[],
+    #     2:[],
+    #     3:[],
+    #     4:[]
+    #      ...
+    #     20:[]
+    # }
+
+    def list_rooms
+      all_room_numbers = @rooms_reservations_hash.keys.sort!
+      return all_room_numbers
+    end
+
+    def list_reservations_for_date(date)
+      res_on_date = []
+      @rooms_reservations_hash.each do |room, reservations|
+        reservations.each do |res|
+          if res.date_range_string_array.include? (date.to_s)
+            this_rooms_reservations = [room]
+            this_rooms_reservations << (res.date_range_string_array.first + " through " + res.date_range_string_array.last)
+          end
+          res_on_date << this_rooms_reservations unless this_rooms_reservations == nil
+        end
+      end
+
+      return res_on_date # array with [[1, "2019-06-10 through 2019-06-12"], [room_num, "date through date"], etc]
+    end
+
+    def ck_avail(ckin, ckout, room_num)
+      # check if specific room is available for this date range
+      # return true/false
+      res_array_to_check = @rooms_reservations_hash[room_num]
+      res_0 = res_array_to_check[0]
+      res_last = res_array_to_check.last
+
+      if res_array_to_check.length == 0
+        return true
+      elsif res_array_to_check.length == 1
+        (res_0.ckout_date <= ckin) || (ckout <= res_0.ckin_date) ? true : false
+      else # there is more than one reservation in the array
+        # check left and right ends of the reservation array
+        return true if (ckout <= res_0.ckin_date)
```

Comment:
Your code maintains your reservations listed in order, later if you take advantage of things Shruti will teach you like binary search some actions like finding a reservation or finding overlapping reservations may be more efficient.  

Nice